packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2940,7 +2940,7 @@ class _RouteEntry extends RouteTransitionRecord {
   /// remove as a result of a page update.
   static const int kDebugPopAttemptLimit = 100;
 
-  static final Route<dynamic> notAnnounced = _NotAnnounced();
+  static final Route<dynamic> notAnnounced = exemptFromRelease(() => _NotAnnounced());
```

Comment:
It would be much cleaner if this could just be:

```dart
  @exemptFromDisposal
  static final Route<dynamic> notAnnounced = _NotAnnounced();
```

Could a transformer read that annotation and turn it into whatever code is necessary to disable the memory tracking?

This would also make it very obvious that none of this logic ends up in un-instrumented release builds because they wouldn't run the transformer.

Reply:
I like the idea.

Reply:
Macros aren't ready yet and using a kernel transformer for this case seems like it would be more magic and complexity than is worth it for this case. 
Can we instead add expandos to objects to tag them as exemptFromRelease?
For example, this code could become
`exemptFromRelease(_NotAnnounced())`
where `exemptFromRelease` is a method that just adds a marker expando to `_NotAnnounced()` indicating it is released. Leak tracking code later checks for the presence of the expando before reporting that the object has not been released. I believe this would also address the footgun concern discussed elsewhere as only this particular object would be exempt rather than all disposables created within the closure.

Reply:
The goal is to exempt ALL disposables. _NotAnnounced have some ChangeNotifier members, that are also not disposed.

It is common pattern: complicated singletone with number of ChangeNotifiers inside, so that widgets could rebuild when some global values, like sign-in status, light/dark theme or connection change. For example, DevTools have number of such controllers. This footgun already exist and we are not sure what to do with it yet.

So, suggestion is not to address it in scope of leak tracker MVP. This will be just one more leak tracker limitation: leak_tracker MVP does not cover web, profile/release, running application, singletones. 

How does it sound?

Reply:
We can rename it to experimentalExemptFromDisposal, to have room to revoke it later